mlflow/transformers.py

```diff
@@ -479,6 +540,61 @@ def log_model(
+                             during inference. These arguments are used exclusively for the case
+                             of loading the model as a ``pyfunc`` Model or for use in Spark.
+                             These values are not applied to a returned Pipeline from a call to
+                             ``mlflow.transformers.load_model()``
+
+                             .. Warning:: If the key provided is not compatible with either the
+                                          Pipeline instance for the task provided or is not a valid
+                                          override to any arguments available in the Model, an
+                                          Exception will be raised at runtime. It is very important
+                                          to validate the entries in this dictionary to ensure
+                                          that they are valid prior to saving or logging.
+
+                             An example of providing overrides for a question generation model:
+
+                             .. code-block:: python
+
+                               from transformers import pipeline, AutoTokenizer
+
+                               task = "text-generation"
+                               architecture = "gpt2"
+
+                               sentence_pipeline = pipeline(
+                                  task=task,
+                                  tokenizer=AutoTokenizer.from_pretrained(architecture),
+                                  model=architecture
+                               )
+
+                               # Validate that the overrides function
+                               prompts = [
+                                 "Generative models are",
+                                 "I'd like a coconut so that I can"
+                               ]
+
+                               # validation of config prior to save or log
+                               inference_config = {
+                                   "top_k": 2,
+                                   "num_beams": 5,
+                                   "max_length": 30,
+                                   "temperature": 0.62,
+                                   "top_p": 0.85,
+                                   "repetition_penalty": 1.15,
+                                   }
+
+                               # Verify that no exceptions are thrown
+                               sentence_generation(prompts, **inference_config)
+
+                               with mlflow.start_run():
+                                 mlflow.transformers.save_model(
+                                   transformers_model=sentence_generation,
+                                   path="/path/for/model",
+                                   task=task,
+                                   inference_config=inference_config
+                                 )
+
```

Comment:
Before releasing this with MLflow 2.3.0, can we make sure that we include documentation for the input / output behavior of pyfunc predict for common pipeline types and pipeline types whose inference behavior may not be intuitive? If possible, can you file an internal ticket to track that?

Reply:
added a ticket and will work on a full mapping of all support pyfunc types input/outputs (making special note of the input types for 2 pipelines that diverge from the transformers API (requiring a Dict input instead of the pseudo-dict string input that is submitted (i.e. `"question: <text> context: <text>"`)